src/frontend/js/forms/update-permalink.ts

```diff
@@ -16,12 +24,59 @@ window.addEventListener("load", () => {
         const url = (document.querySelector('[for="id_title"]') as HTMLElement)
           .dataset.slugifyUrl;
         slugify(url, { title: currentTitle }).then((response) => {
-          document
-            .getElementById("id_slug")
-            .setAttribute("value", response["unique_slug"]);
+          /* on success write response to both slug field and permalink */
+          slugField.value = response["unique_slug"];
+          updatePermalink(response["unique_slug"]);
         });
       });
   }
+
+
+  function toggleSlugMode(){
+    // toggle all permalink buttons
+    let buttonList = document.getElementById("link-container").querySelectorAll("button");
+    for (let button of buttonList) {
+      button.classList.toggle("hidden");
+    }
+    // switch between link and slug field
+    linkContainer.querySelector("a").classList.toggle("hidden");
+    linkContainer.querySelector("div").classList.toggle("hidden");
+  }
+
+
+  function updatePermalink(currentSlug: string){
+    // get complete permalink string
+    let currentLink = linkContainer.querySelector("a").innerHTML;
+    // remove trailing slug from link
+    let updatedLink = currentLink.substr(0, currentLink.lastIndexOf("/")+1);
+    // update only inner html and keep href until form submission
+    linkContainer.querySelector("a").innerHTML = updatedLink.concat(currentSlug);
```

Comment:
```suggestion
    linkContainer.querySelector("a").innerHTML = updatedLink.concat(currentSlug);
    linkContainer.querySelector("a").href = updatedLink.concat(currentSlug);
```
Can you also update the link `href` here? :wink:

Reply:
I'm not sure if this is a good idea, we might break a link that was prior valid, when we update the href attribute before the form is submitted

Reply:
Ah ok, yeah this makes sense to me! I also checked the current WordPress behavior and there it's the same: the link name is the new changed link and the link target is the old, valid link... let's keep it the way it is!